controllers/dspipeline_params.go

```diff
@@ -333,8 +356,19 @@ func (p *DSPAParams) SetupObjectParams(ctx context.Context, dsp *dspa.DataScienc
 		}
 		p.ObjectStorageConnection.AccessKeyID = accessKey
 		p.ObjectStorageConnection.SecretAccessKey = secretKey
-	}
 
+		if dsp.Spec.ObjectStorage.EnableExternalRoute {
+			host, err := p.RetrieveExternalRoute(ctx, client, log)
+			if err != nil {
+				log.Error(err, "Failed to retrieve external route")
```

Comment:
The reason this isn't working is because this introduces a race condition where DSPAParams returns because the Route doesn't exist, and the Route doesn't exist because the DSPAParams is returning before its manifest is applied

Reply:
Thank you @gmfrasca.  Should we use goroutines and channel to handle this scenario ?

Reply:
so you need to update the error handling here such that we dont stop execution if the route doesn't exist, just log the warning or what not and continue on. After that, we need to implement a mechanism to re-run this one the route *does* exist, because the reconciler does not re-queue if no updates to the DSPA CR are made.

Alternatively, we can (probably should) move this handling to AFTER the manifests are applied, since the creation of a Route would re-trigger reconcilation after which we can pick up the metadata appropriately